addons/sale/wizard/sale_make_invoice_advance.py

```diff
@@ -60,12 +60,15 @@ def onchange_advance_payment_method(self):
 
     @api.multi
     def _create_invoice(self, order, so_line, amount):
+        self.ensure_one()
         inv_obj = self.env['account.invoice']
         ir_property_obj = self.env['ir.property']
 
         account_id = False
         if self.product_id.id:
             account_id = self.product_id.property_account_income_id.id
+            if not account_id:
```

Comment:
Why not <code>account_id = this or that</code>?

Reply:
Indeed, using an "or" instead of this additional "if" would be cleaner ;)